server/src/main/java/com/cloud/network/NetworkModelImpl.java

```diff
@@ -593,7 +593,11 @@ public List<? extends Nic> getNics(long vmId) {
     @Override
     public String getNextAvailableMacAddressInNetwork(long networkId) throws InsufficientAddressCapacityException {
         NetworkVO network = _networksDao.findById(networkId);
-        String mac = _networksDao.getNextAvailableMacAddress(networkId, MACIdentifier.value());
+        Integer zoneIdentifyer = MACIdentifier.value();
```

Comment:
I was also wondering if the value of MACIdentifier makes some difference.
Do you think the issue can be fixed by setting the value ? @DaanHoogland

Reply:
I do wonder if 
```
        final SequenceFetcher fetch = SequenceFetcher.getInstance();
        long seq = fetch.getNextSequence(Long.class, _tgMacAddress, networkConfigId);
```
reliably generates a unique mac though. Even after a restart I mean. I haven´t found the persisting of a aequence yet.